### PR TITLE
Keycloak 1.8 integration

### PR DESCRIPTION
**Pull request by romartin**
This PR add support for latest Keylocak 1.8.0.Final. This is the PR for 0.8.x branch, as it hurries for the kie 6.3.x build and qe testing.  I will send the PR for master branch in a while.
Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
NOTE: There are no changes on the kie-wb-distributions repository expected for this update. Only uf-extensions is affected.

File: uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/BaseKeyCloakManager.java [hunk 1]
```diff
@@ -102,12 +102,20 @@ protected void loadConfig( final ConfigProperties config ) {
     }
 
     protected synchronized Keycloak getKeyCloakInstance() {
-        if (this.keycloak == null) {
+        if ( null == this.keycloak ) {
```
Comment (psiroky) [inline uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/BaseKeyCloakManager.java]: Was it intended? Looks a bit weird :)

Comment (csadilek) [inline uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/BaseKeyCloakManager.java]: Yes, looks like a left-over, @romartin? Also can you add some JavaDoc why this method is synchronized?

**Comment by csadilek** [comment]
@romartin just some minor things, see comments above. Then good to go.

File: uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/BaseKeyCloakManager.java (hunk 1 above)
Comment (romartin) [inline uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/BaseKeyCloakManager.java]: will remove the syncronization here yes, does not makes sense. About the notation of the if... just notation, no matter for me to revert it :)

**Comment by romartin** [comment]
@psiroky @csadilek fixed. thx!

**Comment by csadilek** [comment]
@romartin since reference assignments are atomic, the worst that could happen to you after removing the synchronized block is that Keycloak.getInstance is invoked multiple times (and the field is getting reassigned). If you can live with that (seems ok to me), removing synchronized is fine. Can you please comment.

**Comment by romartin** [comment]
@csadilek @psiroky Closing for now. Let's see which concrete version is required to be present. Thx, will keep u updated

**Pull request closed without merging**